staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -120,9 +120,6 @@ func NewCloudNodeController(
 	nodeStatusUpdateFrequency time.Duration,
 	workerCount int32) (*CloudNodeController, error) {
 
-	eventBroadcaster := record.NewBroadcaster()
-	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "cloud-node-controller"})
```

Comment:
moving the initialization to the Run method, causes the controller to panic if it is instantiated but not initialized :(

Reply:
we've seen multiple regressions from threading context through things ... PRs like this need close review... they're not as mechanical as they appear :-/

Reply:
This is seen in other controllers in 1.31.2 as well, xref https://github.com/kubernetes/kubernetes/issues/128935

Reply:
Only the k8s.io/cloud-provider/controllers/service controller has this problem. k8s.io/cloud-provider/controllers/node is okay because the informer callbacks only populate the work queue and don't need the recorder:

https://github.com/kubernetes/kubernetes/blob/953a3a8aa633f9362705081de3bac20012199c4a/staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go#L145-L148

https://github.com/kubernetes/kubernetes/blob/953a3a8aa633f9362705081de3bac20012199c4a/staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go#L296-L306